opta/loader.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+import os
+from typing import List, Optional
+
+from opta.exceptions import UserErrors
+from opta.layer2 import Layer
+from opta.module_spec import SPEC_NAME, ModuleSpec
+from opta.preprocessor import preprocess_layer
+from opta.utils import schema, yaml
+
+
+class LayerLoader:
+    def from_path(self, path: str) -> Layer:
+        try:
+            with open(path) as f:
+                data = yaml.load(f)
+        except FileNotFoundError:
+            raise UserErrors(f"File {path} not found")
+
+        return self.from_dict(data)
+
+    def from_dict(self, raw: dict) -> Layer:
+        # TODO: Validate config data schema (different validation by version?)
+        preprocess_layer(raw)
+
+        layer = Layer.from_dict(raw)
+
+        # TODO: Validate module inputs
```

Comment:
if we already have a jsonschema, then this should be straight forward, yes?

Reply:
Yes, it should be pretty straight forward, but we do need to bypass some of the validation depending on where we are in the apply process.